includes/MediaWiki/Api/EditEndpoint.php

```diff
@@ -141,6 +142,31 @@ public function needsWriteAccess() {
 		return true;
 	}
 
+	/**
+	 * Returns the base for the response body
+	 * @param Status $status
+	 * @return array
+	 */
+	public function getResponseBody( Status $status ): array {
+		$response = [
+			'success' => $status->isGood(),
+			'errors' => []
+		];
+
+		$errors = $status->getErrors();
+		if ( !empty( $errors ) ) {
+			$response['errors'] = array_map( function ( $errorMessage ) {
+				$msg = ApiMessage::create( $errorMessage );
```

Comment:
I’m a bit uncomfortable with the use of `ApiMessage` here, since as far as I can tell the REST API otherwise wants to use `MessageValue` instead of `Message`… but `Message` is what we get from the `Status`, and the MediaWiki core class to convert between the two (`Converter`) isn’t stable, so this is probably the best we can do.

Reply:
Yeah I wasn't entirely sure what is supposed to format/handle these errors either. The REST API seems to also use these but then converting them to message [when throwing localized HTTP errors](https://github.com/wikimedia/mediawiki/blob/REL1_35/includes/Rest/Handler/ActionModuleBasedHandler.php#L138).